### PR TITLE
feature/1207515694088912 Add storage to sdk

### PR DESCRIPTION
**Pull request by jpgundrum**
Added capability for user to interact with our storage pallet using the sdk.

**WIP**: 
- Need to have discussions about best practices and logic for the implementation
- Add more function testing

### CODE REVIEW

File: packages/sdk/src/modules/storage/index.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+import type { ISubmittableResult } from '@polkadot/types/types';
+import type { CodecHash } from '@polkadot/types/interfaces/runtime/types';
+import { CreateStorageKeysEnum, Address } from '../../types';
+import { createStorageKeys } from '../../utils';
+import { stringToU8a, u8aToHex, hexToString } from '@polkadot/util';
+import { StorageError, ItemTypeError, ItemError, StorageAddressError, StorageSeedError} from '../../utils/errors';
+import type { SDKMetadata} from '../../types';
+import { Base } from '../base';
+
+type AddItemOptions = {
```
Comment (jpgundrum) [inline packages/sdk/src/modules/storage/index.ts:11]: currently have it so only a **string** can be set for itemType and item. Should this be changed to offer more types?

File: packages/sdk/src/modules/storage/index.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+        // check if object is uint8array, if it is not then convert to see how big of storage will be needed
+
+        // checks
+        if (!itemType) throw new ItemTypeError('Item Type name is required');
+        if (!item) throw new ItemError('Item name is required');
+        if (seed !== '') this._checkSeed(seed);
+
+        // convert string to bytes to count before calling extrinsics
+        if (stringToU8a(itemType).length > 64) throw new ItemTypeError('New Item Type cannot be larger than 64 bytes');
```
Comment (jpgundrum) [inline packages/sdk/src/modules/storage/index.ts:89]: performs an error check for itemType and item size to reduce the amount of extrinsics sent to the peaq node

File: packages/sdk/src/modules/storage/index.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                },
+            ]);
+            const item = (await api.query?.['peaqStorage']?.['itemStore'](
+                hashed_key
+            ));
+            if (!item || item.isStorageFallback) return null;
+            // is toHuman acceptable here? What if a simple string is not passed?
+            return {
+                log: `${item.toHuman()}`,
+            };
```
Comment (irediaes) [inline packages/sdk/src/modules/storage/index.ts]: Please return the item just as it was stored on chain.
Don't parse it using `toHuman`.

File: packages/sdk/src/modules/storage/index.ts [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+type UpdateItemOptions = {
+    itemType: string;
+    item: string;
+    seed?: string;
+}
+
+type AddItemResult = {
+    log: string;
```
Comment (irediaes) [inline packages/sdk/src/modules/storage/index.ts]: Why naming it "log"?

"message" should fit better

File: packages/sdk/src/modules/storage/index.ts [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    block_hash: CodecHash;
+    unsubscribe: () => void;
+}
+
+type RemoveItemResult = {
+    log: string;
```
Comment (irediaes) [inline packages/sdk/src/modules/storage/index.ts]: "message" fit better.

File: packages/sdk/src/modules/storage/index.ts [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    unsubscribe: () => void;
+}
+
+type GetItemResult = {
+    log: string;
+}
+
+
+type UpdateItemResult = {
+    log: string;
```
Comment (irediaes) [inline packages/sdk/src/modules/storage/index.ts]: "message" fit better.

File: packages/sdk/src/modules/storage/index.ts [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+type RemoveItemResult = {
+    log: string;
+    block_hash: CodecHash;
+    unsubscribe: () => void;
+}
+
+type GetItemResult = {
+    log: string;
```
Comment (irediaes) [inline packages/sdk/src/modules/storage/index.ts]: "data" fit better.

**Review by irediaes — CHANGES_REQUESTED** [review]

**Pull request merged**